hikari/events/message_events.py

```diff
@@ -274,383 +188,431 @@ def is_webhook(self) -> bool:
         """
         return self.message.webhook_id is not None
 
-
-@base_events.requires_intents(intents.Intents.GUILD_MESSAGES, intents.Intents.PRIVATE_MESSAGES)
-@attr.s(kw_only=True, slots=True, weakref_slot=False)
-class MessageUpdateEvent(MessageEvent, abc.ABC):
-    """Event base for any message update event."""
-
     @property
     @abc.abstractmethod
-    def message(self) -> messages.PartialMessage:
-        """Partial message that was sent in the event.
-
-        !!! warning
-            Unlike `MessageCreateEvent`, `MessageUpdateEvent.message` is an
-            arbitrarily partial version of `hikari.messages.Message`
-            where any field except `id` and `channel_id` may be set to
-            `hikari.undefined.UndefinedType` (a singleton) to indicate
-            that it has not been changed.
+    def message(self) -> messages.Message:
+        """Message that was sent in the event.
 
         Returns
         -------
-        hikari.messages.PartialMessage
-            The partially populated message object that was sent with this
-            event.
+        hikari.messages.Message
+            The message object that was sent with this event.
         """
 
     @property
     def message_id(self) -> snowflakes.Snowflake:
-        # <<inherited docstring from MessageEvent>>.
+        """ID of the message that this event concerns.
+
+        Returns
+        -------
+        hikari.snowflakes.Snowflake
+            The ID of the message that this event concerns.
+        """
         return self.message.id
 
+
+@attr.s(kw_only=True, slots=True, weakref_slot=False)
+@base_events.requires_intents(intents.Intents.GUILD_MESSAGES)
+class GuildMessageCreateEvent(MessageCreateEvent):
+    """Event that is fired when a message is created within a guild.
+
+    This contains the full message in the internal `message` attribute.
+    """
+
+    app: traits.RESTAware = attr.ib(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>
+
+    message: messages.Message = attr.ib()
+    # <<inherited docstring from MessageCreateEvent>>
+
+    shard: shard_.GatewayShard = attr.ib(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>
+
     @property
-    def channel_id(self) -> snowflakes.Snowflake:
-        # <<inherited docstring from MessageEvent>>.
-        return self.message.channel_id
+    def author(self) -> users.User:
+        """Member or user that sent the message.
+
+        Returns
+        -------
+        typing.Union[hikari.users.User, hikari.guilds.Member]
+            The user that sent the message. If the member is cached
```

Comment:
wouldn't that use message_payload["member"] not the cache?

Reply:
could not confirm. Although that seems like a sensible assumption.

Can you pop a suggested fix on this? Im not on a pc rn